src/Sulu/Bundle/AdminBundle/Resources/js/components/Input/types.js

```diff
@@ -26,6 +26,8 @@ export type InputProps<T: ?string | ?number> = {|
     onClearClick?: () => void,
     onIconClick?: () => void,
     onKeyPress?: (key: ?string, event: SyntheticKeyboardEvent<HTMLInputElement>) => void,
+    pattern?: string,
```

Comment:
like the idea 👍  would be nice to include other standard `<input>` attributes such as `maxlength` in the future too

Reply:
The problem with those attributes is, that they will only be validated when the form is submitted. And because the form is never being submitted in sulu, the browser doesn't validate those fields 😬

Reply:
yes, i guess we will need to implement the handling of the attributes by ourselves. but it would be nice to provide the functionality to our users 🙂

Reply:
👍

Reply:
First of all, isn't the `submit` event of the form emitted? I thought I have implemented it that way, exactly for reasons like this (also other code might listen to that event...).

Second, I am not sure if I would implement it like this... AFAIK browsers also show their own error based on these properties, which would then probablyl not match our design. I think the best way forward would be to finally allowing field types resp. content types to have an influence on the generated JSON schema. If the JSON schema is generated as required, the frontend would not even have to be touched.

This approach would also fix both issues #5426 and #3943. So we could also corretly check the mandatory field for e.g. the media selection field type. If that's OK for everybody I would be happy to discuss that in detail.

Reply:
Can be, but the native input validation for the `pattern` was never triggered.

If everybody is fine with your suggestion, I would just pause working on this pr until a solution for self-validation is finished.

Reply:
I think I would not event pass the `pattern` to the `<input>`, but implement the handling and the respective message in our `Input` component. Using the already present JSON schema sounds good to me too, but I feel like it will be harder to output helpful error messages that way. Also I dont have an idea how to implement that in a pleasent way in terms of DX, but I have not thought about it in detail. Happy to talk about this too, it would be nice to get some progress in this area soon!